freqtrade/persistence/models.py

```diff
@@ -636,24 +705,17 @@ def calc_profit_ratio(self, rate: Optional[float] = None,
         return float(f"{profit_ratio:.8f}")
 
     def recalc_trade_from_orders(self):
-        # We need at least 2 entry orders for averaging amounts and rates.
-        if len(self.select_filled_orders('buy')) < 2:
-            # Just in case, still recalc open trade value
-            self.recalc_open_trade_value()
-            return
-
         total_amount = 0.0
         total_stake = 0.0
         for o in self.orders:
             if (o.ft_is_open or
                     (o.ft_order_side != 'buy') or
+                    o.is_fully_realized or
```

Comment:
I've said it before, and will say it again.
tracking `is_fully_realized` is pointless. 

you need to track 2 things - realized profit (profit you already sold) - and open profit (profit/loss that's still open).
Especially when recalculating - you need to look at both and update both - as both realized, as well as non-realized will have changed.

By ignoring realized orders, you're potentially skewing / falsifying the result of subsequent orders - assuming the exit size was not identical to the entry size.

The method of tracking "is_fully_realized" will ONLY work if you always trade identical sizes (you enter in steps of 100, exit in steps of 100) - the moment you mix and match values (enter 100, 100, 200,  exit 120) , this will result in wrong results / calculations.

As such - this is covering an **edge-case** (without properly checking this is said edge-case).

Reply:
It was adjusted to different lots.